### PR TITLE
Resolved all conflcits 

### PR DESCRIPTION
**Pull request by jaxter03**

### CODE REVIEW

File: enclave/src/lib.rs [hunk 1]
```diff
@@ -958,6 +960,86 @@ fn handle_ocex_remove_proxy(
...
+        main_acc.clone().encode().to_base58(), //FIXME @gautham please look into it
+        token,
+        amount
+    );
+
+    match polkadex::check_main_account(main_acc.clone().into()) {
+        Ok(exists) => {
+            if exists == true {
+                match polkadex_balance_storage::withdraw(main_acc.clone(), token.clone(), amount) {
+                    Ok(()) => execute_ocex_release_extrinsic(main_acc.clone(), token, amount, 0), // TODO: How to get nonce?
```
Comment (haerdib) [inline enclave/src/lib.rs:1011]: I've written down my thought in issue #12

File: enclave/src/lib.rs [hunk 2]
```diff
@@ -958,6 +960,86 @@ fn handle_ocex_remove_proxy(
...
+    let signer = ed25519::unseal_pair()?;
+    debug!("Restored ECC pubkey: {:?}", signer.public());
+
+    let xt: Vec<u8> = compose_extrinsic_offline!(
+        signer.clone(),
+        call,
+        nonce, // TODO: Where is this?
+        Era::Immortal,
+        genesis_hash,
+        genesis_hash, // FIXME: Shouldn't this be the latest head?
```
Comment (haerdib) [inline enclave/src/lib.rs:1048]: @brenzi do you know the answer to this question or should I look into it?

File: enclave/src/polkadex_balance_storage.rs [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        as *mut SgxMutex<PolkadexBalanceStorage>;
+    if ptr.is_null() {
+        error!("Pointer is Null");
+        return Err(sgx_status_t::SGX_ERROR_UNEXPECTED);
+    } else {
+        Ok(unsafe { &*ptr })
+    }
+}
+
+pub fn deposit(main_acc: AccountId, token: AssetId, amt: u128) -> SgxResult<()> {
```
Comment (haerdib) [inline enclave/src/polkadex_balance_storage.rs]: Proposal: rename function to lock_storage_and_deposit or aquire_lock_and_deposit or something similiar

File: enclave/src/polkadex_balance_storage.rs [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+    // Acquire lock on balance_storage
+    let mutex = load_balance_storage()?;
+    let mut balance_storage: SgxMutexGuard<PolkadexBalanceStorage> = mutex.lock().unwrap();
+    balance_storage.deposit(token, main_acc, amt)
+}
+
+pub fn withdraw(main_acc: AccountId, token: AssetId, amt: u128) -> SgxResult<()> {
```
Comment (haerdib) [inline enclave/src/polkadex_balance_storage.rs]: Proposal: rename function to lock_storage_and_deposit or aquire_lock_and_deposit or something similiar (could also be load_storage_and_desposit, as long as the additonial steps are named)

File: enclave/src/polkadex_balance_storage.rs [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+            }
+        }
+        None => {
+            error!("Account Id or Asset Id is not available");
+            return Err(sgx_status_t::SGX_ERROR_UNEXPECTED);
+        }
+    }
+}
+
+pub fn get_balances(main_acc: AccountId, token: AssetId) -> SgxResult<Balances> {
```
Comment (haerdib) [inline enclave/src/polkadex_balance_storage.rs]: same as the two above

File: enclave/src/test_polkadex_balance_storage.rs [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    let value_one = Balances::from(100u128, 0u128);
+    let key_two = PolkadexBalanceKey::from(AssetId::POLKADEX, main_account_two);
+    let value_two = Balances::from(100u128, 0u128);
+    balance_storage.storage.insert(key_one.encode(), value_one);
+    balance_storage.storage.insert(key_two.encode(), value_two);
+}
+
+#[allow(unused)]
+pub fn initialize_dummy() {
+    {
```
Comment (haerdib) [inline enclave/src/test_polkadex_balance_storage.rs]: I think those curly brackets are unnecessary left overs and are a little bit confusing.. or are they actually necessary?

File: enclave/src/test_polkadex_balance_storage.rs [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+        create_in_memory_balance_storage();
+    }
+    let mutex = load_balance_storage().unwrap();
+    let mut balance_storage = mutex.lock().unwrap();
+    dummy_map(&mut balance_storage);
+}
+
+#[allow(unused)]
+pub fn test_deposit() {
+    {
```
Comment (haerdib) [inline enclave/src/test_polkadex_balance_storage.rs]: Same as above

**Review by haerdib — CHANGES_REQUESTED** [review]
Minor issues, otherwise it does look good to me

File: enclave/src/test_polkadex_balance_storage.rs (hunk 7 above)
Comment (jaxter03) [reply enclave/src/test_polkadex_balance_storage.rs]: done

File: enclave/src/test_polkadex_balance_storage.rs (hunk 6 above)
Comment (jaxter03) [reply enclave/src/test_polkadex_balance_storage.rs]: done

File: enclave/src/polkadex_balance_storage.rs (hunk 4 above)
Comment (jaxter03) [reply enclave/src/polkadex_balance_storage.rs]: done

File: enclave/src/polkadex_balance_storage.rs (hunk 3 above)
Comment (jaxter03) [reply enclave/src/polkadex_balance_storage.rs]: done

File: enclave/src/polkadex_balance_storage.rs (hunk 5 above)
Comment (jaxter03) [reply enclave/src/polkadex_balance_storage.rs]: done

File: enclave/src/lib.rs (hunk 1 above)
Comment (Gauthamastro) [reply enclave/src/lib.rs:1011]: Sure

Comment (Gauthamastro) [reply enclave/src/lib.rs:1011]: We will fix this in a separate issue

File: enclave/src/lib.rs (hunk 2 above)
Comment (Gauthamastro) [reply enclave/src/lib.rs:1048]: Moving it separate issue.

**Review by Gauthamastro — APPROVED** [review]

**Review by haerdib — APPROVED** [review]
LGTM

File: enclave/src/lib.rs (hunk 2 above)
Comment (brenzi) [reply enclave/src/lib.rs:1048]: I believe this only makes a difference when lifetime isn't `immortal`. There is actually an issue in the api-client 
https://github.com/scs/substrate-api-client/issues/108
that would need to be resolved first, so, for now, genesis is fine

**Pull request merged**